game-app/game-core/src/main/java/games/strategy/triplea/delegate/move/validation/MoveValidator.java

```diff
@@ -82,6 +84,14 @@ public class MoveValidator {
   public static final String TRANSPORT_CANNOT_LOAD_AFTER_COMBAT =
       "Transport cannot load after being in combat";
   public static final String NOT_ALL_UNITS_CAN_BLITZ = "Not all units can blitz";
+  public static final String CANNOT_ATTACK_OUT_OF_CONTESTED_TERRITORY =
+      "Cannot attack out of a contested territory";
+  public static final String CANNOT_BLITZ_OUT_OF_BATTLE_INTO_ENEMY_TERRITORY =
+      "Cannot blitz out of a battle into enemy territory";
+  public static final String NOT_ALL_UNITS_CAN_BLITZ_OUT_OF_EMPTY_ENEMY_TERRITORY =
+      "Not all units can blitz out of empty enemy territory";
```

Comment:
Is there any way to make this more specific regarding which units cannot blitz? Can we make this dynamic perhaps and list the unique unit types that cannot?

Reply:
I'm not going to change anything with this error message since it's existing behavior (I just moved it to a variable), but actually the result object already contains the units that were invalid, since this message is added thus:

```
        result.addDisallowedUnit(NOT_ALL_UNITS_CAN_BLITZ_OUT_OF_EMPTY_ENEMY_TERRITORY, u);
```

With `u` being the unit that's not OK. So it's up to the code that's using the result (e.g. the UI code) to display that better, which would be a separate improvement.